agent/checks/check.go

```diff
@@ -1047,6 +1048,99 @@ func (c *CheckGRPC) Stop() {
 	}
 }
 
+type CheckOSService struct {
+	CheckID       structs.CheckID
+	ServiceID     structs.ServiceID
+	OSService     string
+	Interval      time.Duration
+	Timeout       time.Duration
```

Comment:
I don't see `Timeout` being used anywhere. Is it worth implementing a timeout error? Unlike gRPC or script checks I would assume OS service polling shouldn't take too long and it's not worth enforcing.